server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1439,9 +1513,59 @@ public long getGcDeletesInMillis() {
 
     /**
      * Returns the merge policy that should be used for this index.
-     */
-    public MergePolicy getMergePolicy() {
-        return mergePolicyConfig.getMergePolicy();
+     * @param isTimeIndex true if index contains @timestamp field
+     */
+    public MergePolicy getMergePolicy(boolean isTimeIndex) {
+        String indexScopedPolicy = scopedSettings.get(INDEX_MERGE_POLICY);
+        MergePolicyProvider mergePolicyProvider;
+        if (indexScopedPolicy.equals(TIERED_MERGE_POLICY)) {
+            mergePolicyProvider = tieredMergePolicyProvider;
+        } else if (indexScopedPolicy.equals(LOG_BYTE_SIZE_MERGE_POLICY)) {
+            mergePolicyProvider = logByteSizeMergePolicyProvider;
+        } else if (indexScopedPolicy.equals(DEFAULT_POLICY) || Strings.isEmpty(indexScopedPolicy)) {
+            if (!isTimeIndex) {
+                mergePolicyProvider = tieredMergePolicyProvider;
+            } else {
+                String nodeScopedTimeIndexPolicy = TIME_INDEX_MERGE_POLICY.get(nodeSettings);
+                if (nodeScopedTimeIndexPolicy.equals(TIERED_MERGE_POLICY)) {
+                    mergePolicyProvider = tieredMergePolicyProvider;
+                } else if (nodeScopedTimeIndexPolicy.equals(LOG_BYTE_SIZE_MERGE_POLICY)) {
+                    mergePolicyProvider = logByteSizeMergePolicyProvider;
+                } else if (nodeScopedTimeIndexPolicy.equals(DEFAULT_POLICY) || Strings.isEmpty(nodeScopedTimeIndexPolicy)) {
+                    mergePolicyProvider = tieredMergePolicyProvider;
+                } else {
+                    throw new IllegalArgumentException(
+                        "The "
+                            + IndexSettings.TIME_INDEX_MERGE_POLICY.getKey()
+                            + " has unsupported policy specified: "
+                            + nodeScopedTimeIndexPolicy
+                            + ". Please use one of: "
+                            + DEFAULT_POLICY
+                            + ", "
+                            + TIERED_MERGE_POLICY
+                            + ", "
+                            + LOG_BYTE_SIZE_MERGE_POLICY
+                    );
+                }
+            }
+        } else {
+            throw new IllegalArgumentException(
```

Comment:
Same, this should never happen, we are validating policies on settings level

Reply:
replaced with assertion error